app/src/main/java/lt/vilnius/tvarkau/BaseActivity.java

```diff
@@ -21,4 +57,89 @@ public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         Icepick.saveInstanceState(this, outState);
     }
+
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        //isLoginAnon should only be true if user has explicitly clicked browse anonymously button.
+        if(!SharedPrefsManager.instance(this).getIsUserAnonymous()) {
+            connectToGoogle();
+        }
+    }
+
+
+
+    protected void handleSignInResult(GoogleSignInResult result) {
+        Log.d(this.getPackageName(), "handleSignInResult: " + result.isSuccess());
+        if(result.isSuccess()){
+            mGoogleSignInAccount = result.getSignInAccount();
+        }else {
+            //Logging google+ sign in has failed, let user try to login again.
+            startActivity(new Intent(this, SignInActivity.class));
+        }
+
+    }
+
+
+
+    @Override
+    public void onConnectionFailed(ConnectionResult connectionResult) {
+        Log.d(this.getPackageName(), "onConnectionFailed:" + connectionResult);
+    }
+
+
+    protected void showProgressDialog(){
+        if(mProgressDialog != null){
+            mProgressDialog = new ProgressDialog(this);
+            mProgressDialog.setMessage(getString(R.string.progress_dialog_loading));
+            mProgressDialog.setIndeterminate(true);
+        }
+    }
+    protected void hideProgressDialog() {
+        if(mProgressDialog !=null && mProgressDialog.isShowing()){
```

Comment:
Code formatting is incorrect here. Not a big deal, but... Please set up automatic code formatting to be run on each "Save" command. Otherwise, it will be formatted by someone else and "git blame" history will be removed :/